alphastats/tl/differential_expression_analysis.py

```diff
@@ -0,0 +1,431 @@
+                )
+            if not isinstance(metadata, pd.DataFrame):
+                raise TypeError(
+                    "If grouping_column is provided, 'metadata' must be provided."
+                )
+            grouping_values = metadata[grouping_column]
+            sample_values = metadata[Cols.SAMPLE]
+            group1_samples = [
+                sample
+                for sample, group in zip(sample_values, grouping_values)
+                if group == group1
+            ]
+            group2_samples = [
+                sample
+                for sample, group in zip(sample_values, grouping_values)
+                if group == group2
+            ]
+        if len(group1_samples) == 0:
+            raise ValueError("No samples found for group 1.")
+        if len(group2_samples) == 0:
+            raise ValueError("No samples found for group 2.")
+        return group1_samples, group2_samples
+
+
+class DifferentialExpressionAnalysisTTest(DifferentialExpressionAnalysisTwoGroups):
+    """This class implements the t-test differential expression analysis.
+
+    Examples:
+    >>> mat = pd.DataFrame({
+    ...     'sample1': [1, 2, 3],
+    ...     'sample2': [4, 5, 6],
+    ...     'sample3': [7, 8, 9],
+    ...     'sample4': [10, 11, 12]
+    ... }, index=['gene1', 'gene2', 'gene3']).T
+    >>> dea = DifferentialExpressionAnalysisTTest(mat)
+    >>> result = dea.perform(
+    ...     group1_samples = ['sample1', 'sample2'],
+    ...     group2_samples = ['sample3', 'sample4'],
+    ...     is_log2_transformed = True,
+    ...     test_type = "independent",
+    ...     fdr_method = "fdr_bh",
+    ... )
+    >>> result.columns.tolist()
+    ['p-value', 'log2(fold change)', 'q-value']
+    >>> result.index.tolist()
+    ['gene1', 'gene2', 'gene3']
+    """
+
+    def _extend_validation(
+        self,
+        test_type: str,
+        fdr_method: str,
+        **kwargs,
+    ):
+        """Validates the input and parameters for the t-test differential expression analysis.
+
+        Parameters:
+        test_type (str): The test function to use, independent for scipy.stats.ttest_ind or paired for scipy.stats.ttest_rel.
+        fdr_method (str): The FDR method to use, 'fdr_bh' or 'bonferroni'.
+        **kwargs (dict): Additional arguments passed to perform.
+        """
+        super()._extend_validation(**kwargs)
+        if test_type not in [
+            DeaTestTypes.INDEPENDENT,
+            DeaTestTypes.PAIRED,
+        ]:
+            raise ValueError(
+                f"test_type must be either '{DeaTestTypes.INDEPENDENT}' for scipy.stats.ttest_ind or '{DeaTestTypes.PAIRED}' for scipy.stats.ttest_rel."
+            )
+        if fdr_method not in ["fdr_bh", "bonferroni"]:
+            raise ValueError("fdr_method must be one of 'fdr_bh', 'bonferroni'.")
+
+    def _perform(
+        self,
+        is_log2_transformed: bool,
```

Comment:
`is_log2_transformed` is an attribute of the `input_data` and should be set together with it .. otherwise, this added freedom allows for wrong calculations